contracts/extensions/VotingReputation.sol

```diff
@@ -0,0 +1,768 @@
+  }
+
+  function getPollState(uint256 _pollId) public view returns (PollState) {
+    Poll storage poll = polls[_pollId];
+    uint256 requiredStake = getRequiredStake(_pollId);
+
+    // If executed, we're done
+    if (poll.executed) {
+      return PollState.Executed;
+
+    // Not fully staked
+    } else if (
+      poll.stakes[YAY] < requiredStake ||
+      poll.stakes[NAY] < requiredStake
+    ) {
+      // Are we still staking?
+      if (now < poll.lastEvent + STAKE_PERIOD) {
+        return PollState.Staking;
+      // If not, did the YAY side stake?
+      } else if (poll.stakes[YAY] == requiredStake) {
+        return PollState.Executable;
+      // If not, was there a prior vote we can fall back on?
+      } else if (poll.votes[NAY] > 0 || poll.votes[YAY] > 0) {
+        return PollState.Executable;
+      // Otherwise, the poll failed
+      } else {
+        return PollState.Failed;
+      }
+
+    // Fully staked, check for any folds
+    } else if (poll.responses[YAY] == Response.Fold) {
+      return PollState.Failed;
+    } else if (poll.responses[NAY] == Response.Fold) {
+      return PollState.Executable;
+
+    // Do we need to keep waiting?
+    } else if (
+      now < poll.lastEvent + STAKE_PERIOD &&
+      (poll.responses[YAY] == Response.None ||
+      poll.responses[NAY] == Response.None)
+    ) {
+      return PollState.Staking;
+
+    // Fully staked, no folds, go to a vote
+    } else {
+
+      // Infer the right timestamp (since only updated if both parties respond)
+      uint256 lastEvent = poll.lastEvent + ((
+        poll.responses[YAY] == Response.None ||
+        poll.responses[YAY] == Response.None
+      ) ? STAKE_PERIOD : 0);
+
+      if (now < lastEvent + VOTE_PERIOD) {
+        return PollState.Voting;
+      } else if (now < lastEvent + (VOTE_PERIOD + REVEAL_PERIOD)) {
+        return PollState.Reveal;
+      } else if (now < lastEvent + (VOTE_PERIOD + REVEAL_PERIOD + STAKE_PERIOD)) {
+        return PollState.Closed;
+      } else {
+        return PollState.Executable;
+      }
+    }
+  }
+
+  // Internal functions
+
+  function createPoll(
+    address _target,
+    bytes memory _action,
+    uint256 _skillId,
+    bytes memory _key,
+    bytes memory _value,
+    uint256 _branchMask,
+    bytes32[] memory _siblings
+  )
+    internal
+  {
+    pollCount += 1;
+    polls[pollCount].lastEvent = now;
+    polls[pollCount].rootHash = colonyNetwork.getReputationRootHash();
+    polls[pollCount].skillId = _skillId;
+    polls[pollCount].skillRep = checkReputation(pollCount, address(0x0), _key, _value, _branchMask, _siblings);
+    polls[pollCount].target = _target;
+    polls[pollCount].action = _action;
+
+    emit PollCreated(pollCount, _skillId);
+  }
+
+  function getVoteSecret(bytes32 _salt, bool _vote) internal pure returns (bytes32) {
+    return keccak256(abi.encodePacked(_salt, _vote));
+  }
+
+  function toInt(bool _vote) internal pure returns (uint256) {
+    return _vote ? YAY : NAY;
```

Comment:
Primarily because I liked being able to use them to index into storage arrays. Although admittedly I don't love the `toInt` everywhere... I kept trying to take it out but ultimately I liked storing everything in two-member arrays. Open to alternatives.

Reply:
Ok, let's make them `uint256` and throw in a few data-validating `requires`.